api.ts

```diff
@@ -2249,6 +2322,37 @@ export function makeSyncTable<
       }
     : undefined;
 
+  const executeGetPermissions = wrappedExecuteGetPermissions
+    ? async function execGetPermissions(
+        params: ParamValues<ParamDefsT>,
+        request: ExecuteGetPermissionsRequest<K, L, SchemaDefT>,
+        context: GetPermissionExecutionContext,
+      ) {
+        const result = await wrappedExecuteGetPermissions(params, request, context);
+        const {permissions} = result;
+        const permissionCountByRow: {[rowId: string]: number} = permissions.reduce(
+          (acc: {[rowId: string]: number}, permission: Permission) => {
+            acc[permission.rowId] = (acc[permission.rowId] || 0) + 1;
+            return acc;
+          },
+          {},
+        );
+
+        const oversizedRowIds = Object.entries(permissionCountByRow)
+          .filter(([_rowId, count]) => count > MaxPermissionsPerRow)
+          .map(([rowId, _count]) => rowId);
+
+        if (oversizedRowIds.length > 0) {
+          throw new Error(
+            `Objects with ids: ${oversizedRowIds.join(
+              ', ',
+            )} returned more permissions than the maximum allowed of ${MaxPermissionsPerRow} per object`,
```

Comment:
FWIW I believe a pack maker using the CLI can route around the makeSyncTable function, so this limit will need to be enforced on the server too.